src/target/riscv/batch.c

```diff
@@ -122,19 +134,29 @@ int riscv_batch_run(struct riscv_batch *batch, bool resets_delays,
 
 	if (bscan_tunnel_ir_width != 0) {
 		/* need to right-shift "in" by one bit, because of clock skew between BSCAN TAP and DM TAP */
-		for (size_t i = 0; i < batch->used_scans; ++i) {
+		for (size_t i = start_i; i < batch->used_scans; ++i) {
 			if ((batch->fields + i)->in_value)
 				buffer_shr((batch->fields + i)->in_value, DMI_SCAN_BUF_SIZE, 1);
 		}
 	}
 
-	for (size_t i = 0; i < batch->used_scans; ++i)
+	for (size_t i = start_i; i < batch->used_scans; ++i)
 		riscv_decode_dmi_scan(batch->target, batch->idle_count, batch->fields + i,
```

Comment:
If you like - this name would make it more clear what is happening:

```suggestion
		riscv_log_dmi_scan(batch->target, batch->idle_count, batch->fields + i,
```

Reply:
Addressed.